backend/src/useCases/objects/files.ts

```diff
@@ -149,25 +149,33 @@ const retrieveAndReassembleFile = async (
     return Readable.from(chunkData)
   }
 
-  const CHUNK_SIZE = 100
+  const SIMULTANEOUS_CHUNKS = 100
+  let currentIndex = 0
   return new Readable({
-    read: async function () {
-      for (let i = 0; i < metadata.chunks.length; i += CHUNK_SIZE) {
-        const chunks = metadata.chunks.slice(i, i + CHUNK_SIZE)
+    async read() {
+      if (currentIndex >= metadata.chunks.length) {
+        this.push(null)
+        return
+      }
+
+      const endIndex = currentIndex + SIMULTANEOUS_CHUNKS
+      const chunks = metadata.chunks.slice(currentIndex, endIndex)
+      currentIndex = endIndex
+
+      try {
         const chunkedData = await Promise.all(
           chunks.map((chunk) => NodesUseCases.getChunkData(chunk.cid)),
         )
 
         if (chunkedData.some((e) => e === undefined)) {
-          throw new Error('Chunk not found')
+          this.destroy(new Error('Chunk not found'))
+          return
         }
 
-        const canContinue = this.push(Buffer.concat(chunkedData.map((e) => e!)))
-        if (!canContinue) {
-          await new Promise((resolve) => this.once('drain', resolve))
-        }
```

Comment:
Is there a risk in removing back pressure handling?

Reply:
I don't think so since we're limiting the simultaneous batches (~6.4MB) but I'll implement it anyways just in case we'd want to increase or setup an environment variable for this one.

Reply:
Updated